cmd/internal.go

```diff
@@ -286,3 +309,29 @@ func makeColumnMap(columns []pipeline.Column) map[string]string {
 	}
 	return columnMap
 }
+
+// parseLineageRecursively processes the lineage of an asset and its upstream dependencies recursively.
+func parseLineageRecursive(pipeline *pipeline.Pipeline, asset *pipeline.Asset) error {
+	if err := ParseLineage(pipeline, asset); err != nil {
+		return err
+	}
+
+	if len(asset.Columns) == 0 {
+		for _, upstream := range asset.Upstreams {
+			upstreamAsset := pipeline.GetAssetByName(upstream.Value)
+			if upstreamAsset == nil {
+				continue
+			}
+
+			if err := parseLineageRecursive(pipeline, upstreamAsset); err != nil {
+				return err
+			}
+		}
+
+		if err := ParseLineage(pipeline, asset); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
I would have liked these to modify the assets within the `pipeline` object directly, otherwise they will all be lost since the pipeline is not being modified, am I wrong? 

in the end, I think we should design this to be a method like: `pipeline.PropagateColumns()` where it would fill out all the missing columns and modify itself or return a new pipeline object where all the columns are filled out. that way, we could use the same functionality outside of Bruin CLI as well.